index.js

```diff
@@ -48,15 +48,32 @@ var pify = module.exports = function (obj, P, opts) {
 		return opts.include ? opts.include.some(match) : !opts.exclude.some(match);
 	};
 
-	var ret = typeof obj === 'function' ? function () {
-		if (opts.excludeMain) {
-			return obj.apply(this, arguments);
-		}
+	var ret;
```

Comment:
I disagree with the often-unnecessary object instantiation, but not strongly.  If the project maintainers prefer your style, I'll update it.